### PR TITLE
Feature/mbl 14846 edit syllabus

### PR DESCRIPTION
**Pull request by balintbartok**

### CODE REVIEW

**Comment by inst-danger** [comment]
<details><summary>Student.ipa</summary>
  <a href="https://app.bitrise.io/artifact/55094380/p/88c943fe33ffa7be509e75263537f1d2">
    
  </a>
</details>
<details><summary>Teacher.ipa</summary>
  <a href="https://app.bitrise.io/artifact/55094395/p/5efd40fb5089b490ffa3de1300e2df47">
    
  </a>
</details>
<details><summary>Parent.ipa</summary>
  <a href="https://app.bitrise.io/artifact/55094357/p/ee44a37a51bdaa6aa2fb102a98fd48a7">
    
  </a>
</details>

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Build failed, skipping coverage check
  </td>
    </tr>
  </tbody>
</table>



<details><summary>:x: Swift lint</summary>


    ❌ /Users/vagrant/git/Core/Core/Syllabus/SyllabusEditorView.swift:119:1: Line Length Violation: Line should be 200 characters or less: currently 214 characters (line_length)
    /Users/vagrant/git/Core/Core/Syllabus/SyllabusEditorView.swift:28:31: warning: Redundant Optional Initialization Violation: Initializing an optional variable with nil is redundant. (redundant_optional_initialization)
    [student]
</details>
<details><summary>:x: XCTest failed: CoreTests/PageViewEventRequestManagerTests/testRetrievePandataEndpointInfo</summary>


    Assertion Failure at PageViewEventRequestManagerTests.swift:68: Asynchronous wait failed: Exceeded timeout of 5 seconds, with unfulfilled expectations: "events added".
</details>


[MBL-14846](https://instructure.atlassian.net/browse/MBL-14846)

#### Affected Apps: Teacher

#### Release Note: 
none - under feature flag
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Core/Core/Syllabus/SyllabusEditorView.swift [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    }
+}
+
+#if DEBUG
+struct SyllabusEditorView_Previews: PreviewProvider {
+    static var previews: some View {
+        SyllabusEditorView(context: .course("1"), courseID: "1")
+    }
+}
+#endif
```
Comment (vanwagonet) [inline Core/Core/Syllabus/SyllabusEditorView.swift:134]: I love how little code was needed to make this form.

File: Core/CoreTests/Syllabus/SyllabusTabViewControllerTests.swift [hunk 2]
```diff
@@ -67,4 +67,16 @@ class SyllabusTabViewControllerTests: CoreTestCase {
...
+    func testSummaryAppear() {
+        api.mock(controller.course, value: .make(syllabus_body: "not empty"))
+        api.mock(controller.settings, value: .make(syllabus_course_summary: false))
+        controller.view.layoutIfNeeded()
+        controller.viewWillAppear(false)
+
+        XCTAssertEqual(controller.menu?.numberOfItems(inSection: 0), 1)
+        api.mock(controller.settings, value: .make(syllabus_course_summary: true))
+        controller.settings.refresh(force: true)
+        XCTAssertEqual(controller.menu?.numberOfItems(inSection: 0), 2)
```
Comment (vanwagonet) [inline Core/CoreTests/Syllabus/SyllabusTabViewControllerTests.swift:80]: strong work adding tests for the fix there

**Review by vanwagonet — COMMENTED** [review]
Code looks great. QAing now.

**Review by vanwagonet — APPROVED** [review]
QA+1 Works great!

**Pull request merged**